core_backend/app/routers/question_answer.py

```diff
@@ -43,103 +45,108 @@ async def llm_response(
     LLM response creates a custom response to the question using LLM chat and the
     most similar embeddings to the user query in the vector db.
     """
+    (
+        user_query_db,
+        user_query_refined,
+        response,
+    ) = await get_user_query_and_response(user_query, asession)
 
-    feedback_secret_key = generate_secret_key()
-
-    user_query_db = await save_user_query_to_db(
-        asession, feedback_secret_key, user_query
-    )
+    response = await get_llm_answer(user_query_refined, response, qdrant_client)
+    await save_query_response_to_db(asession, user_query_db, response)
 
-    user_query_refined = UserQueryRefined(
-        **user_query.model_dump(), query_text_original=user_query.query_text
-    )
+    return response
 
-    response = UserQueryResponse(
-        query_id=user_query_db.query_id,
-        content_response=None,
-        llm_response="Sorry, we cannot do that.",
-        feedback_secret_key=feedback_secret_key,
-    )
 
-    # -- pre-process input
-    if not input_is_safe(user_query_refined):
-        return response
-
-    user_query_refined = identify_language(user_query_refined)
-    if user_query_refined.original_language == IdentifiedLanguage.UNKNOWN:
-        response.llm_response = (
-            "Sorry, we are unable to understand your question. "
-            "Please rephrase your question and try again."
-        )
+@classify_safety
+@paraphrase_question
+@translate_question
+@identify_language
```

Comment:
It seems like `translate_question` already identifies language in its logic. Since language identification should always precede translation (afaik) should we remove the identify_langauge decorator altogether?

Reply:
Saw your comment. We can keep the decorator for other usage that doesn't require translation I suppose! But we should remove it from this line?

Reply:
I left it in here to test and show how it works. When deploying, we have an option to drop it.

Reply:
You're right. I'm causing more confusion that making things simpler. I'll clean it up so we are not calling it twice

Reply:
Fixed!